### PR TITLE
[Benchmark] Add Gumbel sampler benchmark for speculative decoding

### PR DESCRIPTION
**Pull request by yinli-systems**
## Summary

Adds a targeted benchmark for the V1 GPU worker Gumbel-max sampler used by Model Runner V2 sampling and Eagle-style speculative decoding.

The default shapes mirror the regression report in #40755:

- `num_tokens=40`, `num_reqs=40`, `vocab_size=151936`
- `num_tokens=240`, `num_reqs=40`, `vocab_size=151936`

The benchmark covers processed-logits writeback by default and can also compare:

- sample-only vs processed-logits writeback via `--include-sample-only`
- default fp32 Gumbel noise vs opt-in fp64 via `--include-fp64`
- scalar draft-step output columns vs per-token output columns via `--scalar-output-col`

For `num_tokens > num_reqs`, the default uses per-token output columns to avoid repeated writes to the same request/column slot during a multi-draft-token microbenchmark. `--scalar-output-col` is still available for measuring the current scalar draft-step path directly.

## Testing

- `PYTHONPYCACHEPREFIX=/private/tmp/vllm-pycache /usr/bin/python3 -m py_compile benchmarks/kernels/benchmark_gumbel_sample.py`
- `git diff --check`
- `UV_CACHE_DIR=/private/tmp/uv-cache-vllm UV_TOOL_DIR=/private/tmp/uv-tools-vllm uvx ruff check benchmarks/kernels/benchmark_gumbel_sample.py`
- `UV_CACHE_DIR=/private/tmp/uv-cache-vllm UV_TOOL_DIR=/private/tmp/uv-tools-vllm uvx ruff format --check benchmarks/kernels/benchmark_gumbel_sample.py`

I did not run the benchmark locally because this machine does not have CUDA/H800 access.

### CODE REVIEW

**Comment by yinli-systems** [comment]
Hi maintainers, could you please add the `ready` label if this benchmark PR looks reasonable to run CI? It adds a targeted Gumbel sampler benchmark for #40755 shapes and keeps the PR benchmark-only rather than claiming a kernel fix. Local checks passed: py_compile, ruff check/format, and git diff --check. Thanks!

**Comment by yinli-systems** [comment]
The current `pre-run-check` failure is the repository gate rather than a test failure: it requires a maintainer-applied `ready` or `verified` label, or 4+ merged PRs from the author. Could a maintainer please add the appropriate label if this benchmark PR is worth running through CI?
